088-rebalance-progress-status

```diff
@@ -0,0 +1,328 @@
+
+## Proposal
+
+This proposal would extend the status section of the `KafkaRebalance` custom resource to include a “progress” section that displays information related to ongoing partition rebalances.
+
+In this “progress” section, we include the following fields:
+
+- estimatedTimeToCompletion: The minimum estimated amount time it will take in minutes until partition rebalance is complete. 
+- percentageComplete: The percentage of the partition rebalance that is completed e.g. values in the range [0-100]%
+- rebalanceProgressConfigMap: The ConfigMap where “non-verbose” JSON payload from Executor State from CruiseControlState endpoint is stored.
+
+### Supported KafkaRebalance States
+
+For initial implementation we will focus on including the “progress” section only in the following KafkaRebalance states:
+
+- “Rebalancing” 
+- "Stopped"
+
+These are the states where this progress information will be able to be most accurately calculated and most useful for users. 
+We could provide the “progress” section for other states as well such as the “ProposalReady” and “Ready” states but it is not completely necessary, nor is it trivial. 
+Further explanation as to why that is and why it should be saved as a future improvement is explained in the Future Improvements section near the bottom of this proposal.
+
+All information required for estimating the values of  “estimatedTimeToCompletion” and  “percentageComplete” fields can be derived from either Cruise Control server configurations or CruiseControlState endpoint. 
+That being said, the method of estimation for these fields depends on the state of the KafkaRebalance resource.
+
+#### estimatedTimeToCompletion
+
+##### Rebalancing
+
+```
+rate = (finishedDataMovement)/(<task_trigger_time> - <current_time>)
+
+estimatedTimeToCompletion = (totalDateToMove-finishedDataMovement) / (rate)
+```
+
+##### Stopped
+
+Once a rebalance has been stopped, it cannot be completed. 
+Therefore, there is no “estimationTimeToCompletion” for a stopped rebalance, so we set estimatedTimeToCompletion = null to emphasize this. 
+The `KafkaRebalance` resource must be refreshed and the progress section overwritten with the next state change. 
+
+```
+estimatedTimeToCompletion = null
+```
+
+
+#### percentageComplete
+
+##### Rebalancing
+
+```
+percentageComplete = (finishedDataMovement/totalDataToMoveMB)%
+```
+
+##### Stopped
+
+Once a rebalance has been stopped, it cannot be completed. The `KafkaRebalance` resource must be refreshed and the progress section overwritten with that change. That being said, before the `KafkaRebalace` resource is deleted or “refreshed”, the percentageComplete information will still be of value to users. 
+
+```
+percentageComplete = (finishedDataMovement/totalDataToMoveMB)
+```
+
+#### rebalanceProgressConfigMap
+
+Will only be present in “Rebalancing” and “Stopped” states.
+
+The enhanced `KafkaRebalance` resource would include the following in its status section
```

Comment:
Is this a new section or part of the ConfigMap?

Reply:
This is to show what the `progress` section would look like in the `KafkaRebalance` resource